### PR TITLE
fix(Unpooled Buffers): Fixed wrong buffer size after a load

### PR DESCRIPTION
**Pull request by keyseven123**
## Purpose of the Change and Brief Change Log
This PR fixes a bug that would return the wrong buffer size after a load. The main culprit was that we align the buffer size but return the `neededSize`.

## Verifying this change
This change is tested by the newly added test.

### CODE REVIEW

File: nes-memory/tests/ChildBufferTests.cpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+#include <ctime>
+#include <random>
+
+#include <Runtime/BufferManager.hpp>
+#include <Runtime/TupleBuffer.hpp>
+#include <Util/Logger/Logger.hpp>
+#include <gtest/gtest.h>
+
+/// Testing if the buffer size stays the same during a store --> load
+TEST(ChildBufferTests, StoreAndLoadChildBuffer)
```
Comment (ThanGeo) [inline nes-memory/tests/ChildBufferTests.cpp]: On top of the random sizes, i would add one or two hardcoded tests more where you check a) small, odd values for the sizes (e.g. 1 byte, 7 bytes, 63 bytes etc) and b) exact "power of 2" values (e.g. 8, 16, 32, 64 etc).

That would make the test more deterministic against edge cases
